internal/core/platform.rs

```diff
@@ -347,3 +347,32 @@ impl From<NonZeroPhysicalSize> for PhysicalSize {
         Self::new(size.width.get(), size.height.get())
     }
 }
+
+/// This trait describes the interface GPU accelerated renderers in Slint require to render with OpenGL.
+///
+/// It serves the purpose to ensure that the OpenGL context is current before running any OpenGL
+/// commands, as well as providing access to the OpenGL implementation by function pointers.
+///
+/// # Safety
+///
+/// This trait is unsafe because an implementation of get_proc_address could return dangling
+/// pointers. In practice an implementation of this trait should just forward to the EGL/WGL/CGL
+/// C library that implements EGL/CGL/WGL.
+#[cfg(feature = "std")]
+#[allow(unsafe_code)]
+pub unsafe trait OpenGLInterface {
+    /// Ensures that the OpenGL context is current when returning from this function.
+    fn ensure_current(&self) -> Result<(), Box<dyn std::error::Error>>;
+    /// This function is called by the renderers when all OpenGL commands have been issued and
+    /// the back buffer is reading for on-screen presentation. Typically implementations forward
+    /// this to platform specific APIs such as eglSwapBuffers.
+    fn swap_buffers(&self) -> Result<(), Box<dyn std::error::Error>>;
+    /// This function is called by the renderers when the surface needs to be resized, typically
+    /// in response to the windowing system notifying of a change in the window system.
+    /// For most implementations this is a no-op, with the exception for wayland for example.
+    fn resize(&self, size: NonZeroPhysicalSize) -> Result<(), Box<dyn std::error::Error>>;
+    /// Returns the address of the OpenGL function specified by name, or a null pointer if the
+    /// function does not exist.
+    #[cfg(not(target_arch = "wasm32"))]
```

Comment:
I wonder if we shouldn't keep this funciton enabled on webasm even if it is not called.
otherwise it might be harder for the user to implement this in a cross-platform way.

Reply:
I think there's a compile error with it, but I can try :)

Reply:
Ok, this compiles also for wasm, but the function really doesn't make sense on wasm. The entire trait doesn't make sense on it though, so that's an argument in favor of just keeping it there compiling.